lib/Parse/ParseExpr.cpp

```diff
@@ -2753,17 +2753,49 @@ void Parser::ParseBlockId(SourceLocation CaretLoc) {
 }
 
 bool Parser::StartsBoundsExpression(Token &tok) {
-  if (Tok.getKind() == tok::identifier) {
+  if (tok.getKind() == tok::identifier) {
     IdentifierInfo *Ident = Tok.getIdentifierInfo();
     return (Ident == Ident_byte_count || Ident == Ident_count ||
             Ident == Ident_bounds);
   }
   return false;
 }
 
+bool Parser::StartsInteropTypeAnnotation(Token &tok) {
+  switch (tok.getKind()) {
```

Comment:
Does this mean that the type annotation can't contain any qualifiers (e.g. `const`)?  I'm wondering how confusing that will be to see something like:

    void some_func(const int *p : _Ptr<int>);

It's sort of the "safe" revisionist type for `p` and sort of not.